### PR TITLE
Fix output format

### PR DESCRIPTION
**Pull request by Wolf480pl**
Fix compatibility with OITT output format, and kill reason tracking in general.

- Track machine-readable kill reason (RV/RE/TLE/etc) in OutputBuilder
- Don't confuse exit status with kill signal in OutputBuilder
- Fix compatibility with OITT output format by mangling the exit code reported in the first field of the result line the same way OITT mangles it

### CODE REVIEW

File: src/printer/OutputBuilder.h [hunk 1]
```diff
@@ -7,13 +7,29 @@ namespace printer {
 
 class OutputBuilder {
 public:
+    enum KillReason {
```
Comment (teqwve) [inline src/printer/OutputBuilder.h]: Maybe use `enum class` (stronger type checking)? Also, `KillReason::OK` seems a little weird to me ;p (maybe use to ExecuteStatus::OK/MLE, or KillReason::NONE?)

File: src/printer/OutputBuilder.h [hunk 2]
```diff
@@ -7,13 +7,29 @@ namespace printer {
...
+    static std::string killReasonName(KillReason reason) {
+        switch (reason) {
+            case OK: return "OK";
+            case RE: return "RE";
+            case RV: return "RV";
+            case TLE: return "TLE";
+            case MLE: return "MLE";
+            case OLE: return "OLE";
+        }
+        return "??";
```
Comment (teqwve) [inline src/printer/OutputBuilder.h]: Why case statement and not lookup table as in CODES (or the other way, e.g. case statements are checked for completeness by compiler)? I'm not sure if we take care about consistency/style here and wheather it matters, but...

**Review by teqwve — COMMENTED** [review]
What do You think?

File: src/printer/OutputBuilder.h (hunk 2 above)
Comment (Wolf480pl) [reply src/printer/OutputBuilder.h]: Can't default to `"??" ` with a lookup table.

Comment (Wolf480pl) [reply src/printer/OutputBuilder.h]: Looks like the funciton is not needed anyway.

File: src/printer/OutputBuilder.h (hunk 1 above)
Comment (Wolf480pl) [reply src/printer/OutputBuilder.h]: If it's `enum class` I get `C99 designator ‘RE’ is not an integral constant-expression`, etc. in the lookup table for every enum value. Adding `static_cast<int>` to each of them fixes it, but looks ugly as fuck, so I'll leave it as a regular enum.

File: src/printer/OITimeToolOutputBuilder.cc [hunk 3]
```diff
@@ -94,7 +87,16 @@ int OITimeToolOutputBuilder::encodeStatusCode() const {
             return CODE_RE_BASE + exitStatus_;
         }
     }
-    return CODES[killReason_];
+
+    switch (killReason_){
+        case KillReason::NONE: return 0;
```
Comment (teqwve) [inline src/printer/OITimeToolOutputBuilder.cc]: Why can't we put the code from if above here?

File: src/printer/OITimeToolOutputBuilder.cc [hunk 4]
```diff
@@ -59,16 +61,43 @@ std::string OITimeToolOutputBuilder::dump() const {
...
         ss << "ok";
     }
 }
 
+int OITimeToolOutputBuilder::encodeStatusCode() const {
+    static const int CODE_SIG_BASE = 0;
+    static const int CODE_RE_BASE = 200;
+
+    if (killReason_ == KillReason::NONE) {
+        // NOTE: order of ifs is important, as nonzero killSignal also sets exitStatus_
```
Comment (teqwve) [inline src/printer/OITimeToolOutputBuilder.cc:83]: How about moving code from this if to case KillReason::None below?

File: src/printer/OITimeToolOutputBuilder.cc (hunk 3 above)
Comment (teqwve) [reply src/printer/OITimeToolOutputBuilder.cc]: Duplicated comment, I can't get used to github's review ui.

File: src/printer/OITimeToolOutputBuilder.cc (hunk 4 above)
Comment (Wolf480pl) [reply src/printer/OITimeToolOutputBuilder.cc:83]: No. It'd make the switch bulky and asymmetric.

File: src/printer/OITimeToolOutputBuilder.cc (hunk 3 above)
Comment (Wolf480pl) [reply src/printer/OITimeToolOutputBuilder.cc]: No, it would be ugly.

Comment (Wolf480pl) [reply src/printer/OITimeToolOutputBuilder.cc]: This switch is a _pure_ function from `KillReason` to `int`.

File: src/printer/OITimeToolOutputBuilder.cc (hunk 4 above)
Comment (teqwve) [reply src/printer/OITimeToolOutputBuilder.cc:83]: Oh, sorry, I haven't noticed if in that else.

**Pull request merged**